content/1_docs/2_cookbook/3_templating/0_understanding-oop/recipe.txt

```diff
@@ -0,0 +1,485 @@
+Title: A brief intro to object oriented programming in PHP
+
+----
+
+Published: 2021-01-11
+
+----
+
+Description: Understanding classes, objects, member functions and common pitfalls
+
+----
+
+Text:
+
+## Intro
+
+When you have already tinkered with Kirby for a while, you are likely to have come across terms like classes, objects or methods. You've probably run into errors like "calling a member function x on null" and have then been asked to "check for an object before you call a method". Or have tried to use `$kirby` or `$site` in a plugin and been told to use `kirby()` and `site()` in this context instead. And then you've followed the advice but never really understood why.
+
+If that is true, then this recipe is for you. We will look into the basics of object oriented programming (OOP) with the goal of making you understand what this is all about.
+
+The Kirby core is written in PHP, a scripting language that still powers a majority of websites.
+
+Like basically all professional PHP applications today, Kirby is based on an OOP (Object Oriented Programming) architecture, which has many advantages over the procedural approach you often find in older PHP programs. The idea behind the OOP architecture is to encapsulate data and the methods that work with this data, in order to produce modular, extensible and maintainable code.
+
+<warning>
+No images, lots of text 😉. But at least some code to play with. So get some coffee or tea.
+</warning>
+
+## Prerequisites
+
+If you want to code along with this recipe, you need the following:
+
+- A web server (ideally on your (link: docs/cookbook/setup/development-environment text: local machine))
+- A current version of PHP
+- A Kirby installation, either (link: docs/guide/quickstart text: Starterkit or Plainkit)
```

Comment:
Why is a Kirby installation required for the tutorial? As far as I can see, all classes work on their own.

Reply:
Oops, you are right, that's a leftover from the first version